### DISCUSSION TITLE
Управление устройствами на базе Linux BusyBox по SSH

### DISCUSSION THREAD

**Discussion by tvideo** [totumonline/totum-ru-issues-and-discussions#112] [Как это сделать?] [answered]
![схема](https://user-images.githubusercontent.com/32614108/163805110-6985766b-72e4-4b9d-a040-675283a6b54a.jpg)

**Описание:**

Есть хостинг `totum`. На этом хостинге нельзя устанавливать доп. пакеты, в частности `SSHPASS`, поэтому был создан VDS на `Ubuntu` и установлен на него `SSHPASS`.
 
Totum по `SSH` по ключу подключается к `VDS (Ubuntu)`, а он, в свою очередь, неинтерактивно подключается к устройству `Linux BusyBox` и запускает находящийся там `.sh` скрипт.

Скрипт, который запускает терминал totum выглядит так:

```
ssh root@80.87.97.50 SSHPASS='root' sshpass -e ssh -o StrictHostKeyChecking=no root@86.57.135.85 -p222 /root/plugin/var/tuxbox/res.sh
```

Скрипт `/root/plugin/var/tuxbox/res.sh` выполняет просто перезапуск одного модуля в `Linux BusyBox`.

**Проблема:**

Если этот скрипт `/root/plugin/var/tuxbox/res.sh` запустить с терминала устройства `Linux BusyBox`, то он работает, все ок.

Если скрипт отправлять из `totuma` по `SSH`:

```
ssh root@80.87.97.50 SSHPASS='root' sshpass -e ssh -o StrictHostKeyChecking=no root@86.57.135.85 -p222 /root/plugin/var/tuxbox/res.sh
```

то на устройстве Linux BusyBox все выполняется верно, но `totum` при этом находится в ожидании (крутится шестеренка), хотя команда на устройстве `Linux` выполнена и выведен `echo`.

**Вопрос:**

Можно ли каким-то образом написать так, что бы `totum` отправив приведенных выше `SSH`, не ждал ответа. 

Допустим 2 сек и закрывал соединение (либо как-то отключался), что бы он не находился в режиме ожидания ответа (крутится шестеренка).

Спасибо

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Попробуйте в конце linux-команды поставить `&` — это означает `отвязать команду от терминала`.

Ток я вижу у вас там как в фильме с Ди-Каприо: команда в команде в команде.

Еще вариант сделать на промежуточном сервере исполняемый .sh файл и тотуму запускать его, а он уже исполняет скрипт на BusyBox.
